### PR TITLE
:sparkles: keyValue / remove by object

### LINKED ISSUE #118: remove with instance object
Right now we can remove instance from `keyValue` per their `key`.
Add the possibility to remove them per reference equality to full instance.

pseudo code:
```js
const = remove = ([object]) => {
  if (typeof object === 'object') {
    // remove per reference equality
  } else {
    // remove per id 
  }
}
```

### PR DESCRIPTION
**Pull request by fabienjuif**
fixes #118 

@EmrysMyrddin This is not really "by reference", because we are looking for the key if the type is `object` but I guess it is ok like that ? WDYT ?

### CODE REVIEW

File: src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap [hunk 1]
```diff
@@ -897,6 +897,65 @@ Object {
 }
 `;
 
+exports[`middlewares/keyValue should remove element by its reference [elm1] 1`] = `
+Object {
+  "action": Object {
+    "payload": Object {
```
Comment (fabienjuif) [inline src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap]: We remove with a single object (elm1)

File: src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap [hunk 2]
```diff
@@ -897,6 +897,65 @@ Object {
...
+      "code": "elm1",
+      "infos": "elm1",
+      "some": "other",
+      "subinfos": Object {
+        "info": "elm1",
+      },
+    },
+    "type": "@@krf/REMOVE>TESTPREFIX>TESTNAME",
+  },
+  "state": Object {
```
Comment (fabienjuif) [inline src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap]: We can see elm1 is not into state anymore

File: src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap [hunk 3]
```diff
@@ -935,6 +994,58 @@ Object {
 }
 `;
 
+exports[`middlewares/keyValue should remove elements by their references [elm1, elm2] 1`] = `
+Object {
+  "action": Object {
+    "payload": Array [
```
Comment (fabienjuif) [inline src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap]: We remove with an array of object (elm1 and elm2)

File: src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap [hunk 4]
```diff
@@ -935,6 +994,58 @@ Object {
...
+        "infos": "elm2",
+        "some": "other",
+        "subinfos": Object {
+          "info": "elm2",
+        },
+      },
+    ],
+    "type": "@@krf/REMOVE>TESTPREFIX>TESTNAME",
+  },
+  "state": Object {
```
Comment (fabienjuif) [inline src/types/keyValue/__snapshots__/keyValue.middleware.spec.js.snap]: We can see elm1 and elm2 are not into state anymore

**Comment by EmrysMyrddin** [comment]
Yes I'm agree with that. It's what I had in mind :-)

**Comment by fabienjuif** [comment]
Nice, then if you can review this PR it would be nice ;)

**Pull request merged**